operatorcontrolplane/v1alpha1/types_conditioncheck.go

```diff
@@ -0,0 +1,150 @@
+	// be resolved. Specify an IP address for host to bypass DNS name lookup.
+	// +kubebuilder:validation:Required
+	// +required
+	TargetEndpoint string `json:"targetEndpoint"`
+}
+
+// +k8s:deepcopy-gen=true
+type PodNetworkConnectivityCheckStatus struct {
+	// Successes contains logs successful check actions
+	// +optional
+	Successes []LogEntry `json:"successes"`
+
+	// Failures contains logs of unsuccessful check actions
+	// +optional
+	Failures []LogEntry `json:"failures"`
+
+	// Outages contains logs of time periods of outages
+	// +optional
+	Outages []OutageEntry `json:"outages"`
+
+	// Conditions summarize the status of the check
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []PodNetworkConnectivityCheckCondition `json:"conditions" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// LogEntry records events
+type LogEntry struct {
+	// Start time of check action.
+	Start metav1.Time `json:"time,omitempty"`
+
+	// Status indicates if the action performed by the check was successful or not.
+	// +required
+	Success bool `json:"status"`
+
+	// Reason for status in a machine readable format.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// Message explaining status in a human readable format.
+	// +optional
+	Message string `json:"message,omitempty"`
+
+	// Latency records how long the action mentioned in the entry took.
+	// +optional
+	Latency metav1.Duration `json:"latency"`
+}
+
+// OutageEntry records time period of an outage
+type OutageEntry struct {
+
+	// Start of outage detected
+	// +kubebuilder:validation:Required
+	// +required
+	Start metav1.Time `json:"start"`
+
+	// End of outage detected
+	// +kubebuilder:validation:Required
+	// +optional
+	End metav1.Time `json:"end"`
+}
+
+// PodNetworkConnectivityCheckCondition represents the overall status of the pod network connectivity.
+// +k8s:deepcopy-gen=true
+type PodNetworkConnectivityCheckCondition struct {
+
+	// Type of the condition
+	// +kubebuilder:validation:Required
+	// +required
+	Type PodNetworkConnectivityCheckConditionType `json:"type"`
+
+	// Status of the condition
+	// +kubebuilder:validation:Required
+	// +required
+	Status metav1.ConditionStatus `json:"status"`
+
+	// Reason for the condition's last status transition in a machine readable format.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// Message indicating details about last transition in a human readable format.
+	// +optional
+	Message string `json:"message,omitempty"`
+
+	// Last time the condition transit from one status to another.
+	// +kubebuilder:validation:Required
+	// +required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+}
+
+type LogEntryReason string
+
+const (
+	DNSResolve      LogEntryReason = "DNSResolve"
+	DNSError        LogEntryReason = "DNSError"
+	TCPConnect      LogEntryReason = "TCPConnect"
+	TCPConnectError LogEntryReason = "TCPConnectError"
```

Comment:
timeout?

Reply:
Error is not always timeout, for example _no route to host_.

Reply:
I think I wanted to say that I miss a timeout value.